### ISSUE TITLE
Please add "maps" to the content db

### ISSUE THREAD

**Issue by CHJ85** [minetest/contentdb#162]
Hi there. Do you think you could add the maps/worlds (submitted in the forum) to the content db?
Especially because some of these games have maps created specifically for those games.

**Comment by rubenwardy** [comment]
The problem is that maps aren't supported by minetest, and ContentDB only hosts things that can be installed by minetest. How should maps work? Should you select the map instead of a map gen in the world create menu?


The current recommended approach is for games to use schematics and mapgen hooks to place pre-built maps

**Comment by CHJ85** [comment]
Well. then just add a disclaimer letting people know that the maps are community created and not officially supported by Minetest.
Then when people download a map, let them choose which game they wish to add it to.
Should be simple enough as it's 1 line in the world.mt file.
There are some really good maps out there on the forums, and I wish people knew about them and world play them. And some of these maps are meant for specific games in mind.
I just think it'd be a fine addition to the game, that's all.

**Comment by benrob0329** [comment]
As of right now, they should install like any other map/world. Updates probably wouldn't work with them, unless they were readonly.

**Comment by CHJ85** [comment]
Right. But how many people really search the forums, let alone are aware of its existence?
And since you're now able to install mods, games and texture packs from within the game, it'd be nice if this also applied to maps/worlds created by the community.

**Comment by appgurueu** [comment]
Updates *would* work, but won't be as simple: 

* Chunks that have not been touched by the player can easily be overwritten with the new data
* Other chunks, modstorage, and custom files would create *merge conflicts*

Much like modifying source code of a downloaded mod poses a problem as well (as it might be overwritten).

**Comment by rubenwardy** [comment]
Idea for implementation:

ContentDB automatically rewrites the zips of the world to use the readonly fall back database. The implication of this is that the readonly sqlite file would be updated when an update is received, and the normal map.sqlite would contain any user modifications. This would allow modifications without losing user changes

However, because Minetest will just save any loaded blocks, this means that updates won't affect places that a user has visited

**Comment by celeron55** [comment]
Do we have an issue about MT needing to have support for pre-made maps? In my mind I see it as basically a mapgen that just copies from another (read-only) world into the one that's active, instead of generating from scratch.

**Comment by rubenwardy** [comment]
> Do we have an issue about MT needing to have support for pre-made maps? In my mind I see it as basically a mapgen that just copies from another (read-only) world into the one that's active, instead of generating from scratch.

We support read-only source databases in a world - you can create another sqlite file and load from it instead of mapgen. My plan for this is to only support sqlite uploads, and do some clever stuff with making the zip archive or installing such that the world goes into the read-only database. This isn't quite the same but would probably work quite well

**Comment by Zughy** [comment]
How about sharing schematics, like a free asset store? I've got this idea while reading this proposal for my minigame lib, where the user wanted to implement an editor to submit new arenas for review => https://gitlab.com/zughy-friends-minetest/arena_lib/-/issues/118

If this is implemented upstream, it'll be way easier

**Comment by Zughy** [comment]
@BuckarooBanzay that's really cool, it'll be so nice to see it implemented on CDB (or similar) by default. Basically, like Minecraft has its own marketplace, we could have ours for schematics

**Comment by CHJ85** [comment]
^ Yes, that would be awesome!
Great work @BuckarooBanzay 👍
